_posts/2020-11-04-differences-between-programming-languages.md

```diff
@@ -0,0 +1,158 @@
+Java is an object-oriented and high-level programming language; and it can run on all platforms and all sorts of hardwares. A data collected on Github shows that Java has been the top popular programming language in web design and web applications. [^2] Java is particularly common in web-based development, as well as the development in video games and mobile applications.
+
+Java's primary principle is:
+> write once, run anywhere (WORA)
+
+### - JavaScript
+JavaScript is high-level and multi-paradigm. It is a client-side programming language, written for the web browser. Unlike Java which objects are class-based, JavaScript objects are prototype-based. [^3] The main difference between Java and JavaScript is the design idea, where JavaScript is more user-friendly and interactive.
+
+## 3. Ruby
+![Ruby](https://i.pinimg.com/564x/eb/9e/3b/eb9e3b7dab09358e7cf13f188f64f9f4.jpg)
+
+Ruby is similar to Python. It is a high-level, general-purpose programming language. Ruby uses garbage collection and it supports multiple paradigms, including functional programming and object-oriented programming. Ruby was released its first version Ruby 1.0 in December, 1996; and now it is on the Ruby 3.0. [^4] Ruby is commonly used in 3D modeling and it is a useful tool to manage and track information.
+
+Ruby is a part of the Ruby on Rails web framework. Ruby uses primary on its website-building framework Ruby on Rails.
+
+### History of Ruby
+Ruby was designed by Yukihiro Matsumoto in Japan. He wrote his early idea of developing Ruby in his post:
+
+>I was talking with my colleague about the possibility of an object-oriented scripting language. I knew Perl (Perl4, not Perl5), but I didn't like it really, because it had the smell of a toy language (it still has). The object-oriented language seemed very promising. I knew Python then. But I didn't like it, because I didn't think it was a true object-oriented language – OO features appeared to be add-on to the language. As a language maniac and OO fan for 15 years, I really wanted a genuine object-oriented, easy-to-use scripting language. I looked for but couldn't find one. So I decided to make it.
+
+## 4. Bash
+![Bash](https://www.fullstackpython.com/img/logos/bash-wide.jpg)
+
+Bash is a scripting language. It is a a command processor and it is a Unix shell. Bash can run in a text editor window where user types commands and runs to make actions. Bash can also execute commands from a shell script file.
+
+Bash was originally released in 1989 written by Brian Fox. The language was designed primarily as a free software replacement for the Bourne shell. [^5]
+
+Bash contains a variety of features, including wildcard matching, piping, control structures, command history and brace expansion, etc. It has both features from the C shell's major advantages and its own features. [^6]
+
+### Bash Script Example
+```
+#!/bin/bash
+n=0
+if [ $n -eq 0 ];
+then
+echo "It is number 0"
+else
+echo "It is not number 0"
+fi
+```
```

Comment:
@DingShen29 Like the above comment, perhaps a brief explanation about what this example code is doing could help.